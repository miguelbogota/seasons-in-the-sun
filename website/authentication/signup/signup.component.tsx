import { TextField } from '@app/ui/text-field';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';

import { type AppSignupCredentials, appSignupCredentialsSchema, useAuthentication } from '../state';

export function Signup() {
  const { signup } = useAuthentication();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<AppSignupCredentials>({
    criteriaMode: 'all',
    resolver: zodResolver(appSignupCredentialsSchema),
  });

  const onSubmit = handleSubmit(async (data) => {
    await signup(data).catch(console.error);
  });

  return (
    <div className="signup">
      <h2>Join</h2>
      <form onSubmit={onSubmit}>
        <TextField
          {...register('username')}
          label="Username"
          placeholder="Ex. johndoe"
          hint={errors.username?.message}
        />

        <TextField {...register('password')} label="Password" />
        {errors?.password?.types &&
          Object.entries(errors.password.types).map(([type, message]) =>
            Array.isArray(message) ? (
              message.map((message) => (
                <>
                  <p key={type}>{message}</p>
                  <br />
                </>
              ))
            ) : (
              <>
                <p key={type}>{message}</p>
                <br />
              </>
            ),
          )}

        <TextField
          {...register('confirmPassword')}
          label="Confirm Password"
          hint={errors.confirmPassword?.message}
        />

        <TextField
          {...register('email')}
          label="Email"
          placeholder="Ex. johndoe@email.com"
          hint={errors.email?.message}
        />

        <button className="button" type="submit">
          Sign up
        </button>
      </form>
    </div>
  );
}
