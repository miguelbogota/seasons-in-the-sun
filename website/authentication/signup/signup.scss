@mixin apply {
  .signup {
    display: block;
  }
}
