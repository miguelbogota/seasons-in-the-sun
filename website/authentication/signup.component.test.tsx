import { render } from '@testing-library/react';

import { Signup } from './signup.component';

describe('Website / Authentication / Signup / Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Signup />);
    expect(container).toMatchSnapshot();
  });
});
