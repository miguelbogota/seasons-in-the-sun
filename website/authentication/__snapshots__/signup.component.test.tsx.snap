// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Website / Authentication / Signup / Component > should match snapshot 1`] = `
<div>
  <div>
    Sign up
  </div>
</div>
`;
